cpp/src/copying/copy_range.cu

```diff
@@ -71,6 +83,65 @@ struct inplace_copy_range_dispatch {
   }
 };
 
+struct out_of_place_copy_range_dispatch {
+  cudf::column_view const& source;
+  cudf::column_view const& target;
+
+  template <typename T>
+  std::enable_if_t<cudf::is_fixed_width<T>(), std::unique_ptr<cudf::column>>
+  operator()(
+      cudf::size_type source_begin, cudf::size_type source_end,
+      cudf::size_type target_begin,
+      rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource(),
+      cudaStream_t stream = 0) {
+    auto p_ret = std::make_unique<cudf::column>(target, stream, mr);
+    if ((!p_ret->nullable()) && source.has_nulls()) {
```

Comment:
@jrhemstad @harrism 
I feel like this should be

```
if ((!p_ret->nullable()) && source.has_nulls(source_begin, source_end)) {
```

and do you guys have any concerns on adding has_nulls(size_type begin, size_type end) to colum_view and making this update?

and if this change is legitimate,

In the previous PR,

@cwharris initiated the discussion about the fill function's behavior when source_begin == source_end and value.is_valid() is false.

https://github.com/rapidsai/cudf/pull/3172/files#diff-1c1b3a4ff058e5cd78c94f6328cc58b5R117

We decided to set the returned column nullable() in this case, but if we consider fill as a special case of copy_range (with an input column returning a constant value whatever the row index is), fill's behavior with source_begin == source_end should be identical with copy_range's behavior with source_begin == source_end; and with the above update (we call has_nulls(source_begin, source_end) instead of just has_nulls()), I think fill should return a non-nullable column in this case. Any concerns on making this change?

Reply:
No concern from me. I think if it's a no-op, the output should be the input.